substrate/client/network/benches/request_response_protocol.rs

```diff
@@ -0,0 +1,240 @@
+	let local_addr = listener.local_addr().unwrap();
+	let port = local_addr.port();
+
+	build_multiaddr!(Ip4(ip), Tcp(port))
+}
+
+pub fn create_network_worker(
+	listen_addr: sc_network::Multiaddr,
+) -> (
+	NetworkWorker<runtime::Block, runtime::Hash>,
+	async_channel::Receiver<IncomingRequest>,
+	Box<dyn NotificationService>,
+) {
+	let (tx, rx) = async_channel::bounded(10);
+	let request_response_config =
+		NetworkWorker::<runtime::Block, runtime::Hash>::request_response_config(
+			"/request-response/1".into(),
+			vec![],
+			MAX_SIZE,
+			MAX_SIZE,
+			Duration::from_secs(2),
+			Some(tx),
+		);
+	let mut net_conf = NetworkConfiguration::new_local();
+	net_conf.listen_addresses = vec![listen_addr];
+	let mut network_config = FullNetworkConfiguration::new(&net_conf, None);
+	network_config.add_request_response_protocol(request_response_config);
+	let (block_announce_config, notification_service) = NonDefaultSetConfig::new(
+		"/block-announces/1".into(),
+		vec![],
+		1024,
+		Some(NotificationHandshake::new(BlockAnnouncesHandshake::<runtime::Block>::build(
+			Roles::from(&Role::Full),
+			Zero::zero(),
+			runtime::Hash::zero(),
+			runtime::Hash::zero(),
+		))),
+		SetConfig {
+			in_peers: 1,
+			out_peers: 1,
+			reserved_nodes: vec![],
+			non_reserved_mode: NonReservedPeerMode::Accept,
+		},
+	);
+	let worker = NetworkWorker::<runtime::Block, runtime::Hash>::new(Params::<
+		runtime::Block,
+		runtime::Hash,
+		NetworkWorker<_, _>,
+	> {
+		block_announce_config,
+		role: Role::Full,
+		executor: Box::new(|f| {
+			tokio::spawn(f);
+		}),
+		genesis_hash: runtime::Hash::zero(),
+		network_config,
+		protocol_id: ProtocolId::from("bench-request-response-protocol"),
+		fork_id: None,
+		metrics_registry: None,
+		bitswap_config: None,
+		notification_metrics: NotificationMetrics::new(None),
+	})
+	.unwrap();
+
+	(worker, rx, notification_service)
+}
+
+async fn run_consistently(size: usize, limit: usize) {
+	let mut received_counter = 0;
+	let listen_address1 = get_listen_address();
+	let listen_address2 = get_listen_address();
+	let (mut worker1, _rx1, _notification_service1) = create_network_worker(listen_address1);
+	let service1 = worker1.service().clone();
+	let (worker2, rx2, _notification_service2) = create_network_worker(listen_address2.clone());
+	let peer_id2 = *worker2.local_peer_id();
+
+	worker1.add_known_address(peer_id2, listen_address2.into());
+
+	let requests = async move {
+		while received_counter < limit {
+			received_counter += 1;
+			let _ = service1
+				.request(
+					peer_id2.into(),
+					"/request-response/1".into(),
+					vec![0; 2],
+					None,
+					IfDisconnected::TryConnect,
+				)
+				.await
+				.unwrap();
+		}
+	};
+	let network1_run = worker1.run();
+	let network2_run = worker2.run();
+	tokio::pin!(requests);
+	tokio::pin!(network1_run);
+	tokio::pin!(network2_run);
+
+	loop {
+		tokio::select! {
+			_ = &mut network1_run => {},
+			_ = &mut network2_run => {},
```

Comment:
The same as with notifications, we better spawn a separate tokio task for every "node".